javalib/src/main/scala/java/nio/channels/MembershipKey.scala

```diff
@@ -0,0 +1,15 @@
+package java.nio.channels
+
+import java.net.{InetAddress, NetworkInterface}
+
+abstract class MembershipKey {
+  def isValid(): Boolean
+  def drop(): Unit
+  // TODO: implement these methods
```

Comment:
Developers choice suggestion here.

Since this class is abstract, the `TODO` could be left abstract.

From reading the JVM 22 doc for MembershipKey, it seems
like `block()` could be implemented as always returning `UnsupportedOperationException`, along with a comment in
the code that  that support should be added. I have not
dealt with net group membership. Do both Windows and Unix-like
support it?  I suspect yes.

Similarly, `unblock` could be implemented as always returning
`IllegalStateException`, along with a comment about in a better
world it would be implemented but in this one since the Key can
never block, it is guaranteed to be in the wrong state.

Both of these use Exceptions documented for the methods.

One could create a 5 minute or less Issue pointing to this PR
and stating that actually supporting MembershipKey was deferred
and still needs, for some definition of "needs", to be done.

That Issue may sink down the priority list, but it will be a reminder
to devos and may be found by users who stumble across the issue
in the wild.

Your thoughts?

Then `MembershipKeyImpl` could implement `block`